crates/sncast/tests/e2e/script/declare.rs

```diff
@@ -41,10 +44,125 @@ async fn test_wrong_contract_name() {
     ];
 
     let snapbox = Command::new(cargo_bin!("sncast"))
-        .current_dir(SCRIPTS_DIR.to_owned() + "/declare/no_contract")
+        .current_dir(SCRIPTS_DIR.to_owned() + "/declare/test_scripts")
         .args(args);
-    snapbox.assert().success().stderr_matches(indoc! {r"
+    snapbox.assert().success().stdout_matches(indoc! {r#"
+        ...
+        ScriptCommandError::ContractArtifactsNotFound(ErrorData { msg: "Mapaaaa" })
+        command: script
```

Comment:
just to make sure, currently we:

- always return success, when the script suceeds (so executes everything from top to bottom)
- return an error, when there is an error (be it script error, sncast error etc)

right? maybe it would be worth it to also have a test that returns an ectual error (eg contract artifact is not found, and because of this status is `error` instead of `success`). thoughts?

Reply:
I think we already have a test with a script error
https://github.com/foundry-rs/starknet-foundry/pull/1707/files#diff-099822b498a0b060310da7d23400256ed8dbc1aa36eb6e861232da1e9d03e3b7L3-L6

https://github.com/foundry-rs/starknet-foundry/pull/1707/files#diff-69c2fe9dd83d3ca7cf89064e00d74238cbce15d4226c49725ed8464264f5f554R20-R35

Reply:
ah ok, so it is added in a PR that will be opened next and is not a draft. that's fine, thanks :+1:

Reply:
yeah, I decided to keep those as drafts so we can focus on the first two PRs for now